src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -448,6 +448,10 @@ private void ConsumeCurrentTokenInto(HttpSyntaxNode node)
                         node = new HttpUrlNode(_sourceText, _syntaxTree);
                         node.Add(ParseEmbeddedExpression());
                     }
+                    else if (CurrentToken is { Kind: HttpTokenKind.Punctuation })
```

Comment:
Why do we enter the while loop in the case where the current token is a punctuation (on line 436 above)? A url should always start with a word (or an embedded expression) should it not?

Reply:
Never mind - I think I understand what is going on here..